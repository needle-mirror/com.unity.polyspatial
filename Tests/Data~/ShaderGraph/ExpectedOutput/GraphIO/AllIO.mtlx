<?xml version ="1.0"?>
<materialx version = "1.38" colorspace = "lin_rec709">
    <constant name = "_Matrix4x4" type = "matrix44">
        <input name = "value" type = "matrix44" value = "1,0,0,0,0,1,0,0,0,0,1,0,0,0,0,1" />
    </constant>
    <constant name = "_Vector4" type = "vector4">
        <input name = "value" type = "vector4" value = "0,1,0,0" />
    </constant>
    <dot name = "_1" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_2" />
    </dot>
    <transformmatrix name = "_2" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_3" />
        <input name = "mat" type = "matrix44" nodename = "_4" />
    </transformmatrix>
    <dot name = "_3" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_Vector4" />
    </dot>
    <transpose name = "_4" type = "matrix44">
        <input name = "in" type = "matrix44" nodename = "_5" />
    </transpose>
    <dot name = "_5" type = "matrix44">
        <input name = "in" type = "matrix44" nodename = "_Matrix4x4" />
    </dot>
    <constant name = "_Vector3" type = "vector3">
        <input name = "value" type = "vector3" value = "1,0,0" />
    </constant>
    <constant name = "_Matrix3x3" type = "matrix33">
        <input name = "value" type = "matrix33" value = "1,0,0,0,1,0,0,0,1" />
    </constant>
    <dot name = "_6" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_7" />
    </dot>
    <transformmatrix name = "_7" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_8" />
        <input name = "mat" type = "matrix33" nodename = "_9" />
    </transformmatrix>
    <dot name = "_8" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_Vector3" />
    </dot>
    <dot name = "_9" type = "matrix33">
        <input name = "in" type = "matrix33" nodename = "_Matrix3x3" />
    </dot>
    <dot name = "_10" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_11" />
    </dot>
    <combine4 name = "_11" type = "vector4">
        <input name = "in1" type = "float" nodename = "_12" />
        <input name = "in2" type = "float" nodename = "_13" />
        <input name = "in3" type = "float" nodename = "_14" />
        <input name = "in4" type = "float" nodename = "_15" />
    </combine4>
    <dot name = "_12" type = "float">
        <input name = "in" type = "float" nodename = "_Float" />
    </dot>
    <dot name = "_13" type = "float">
        <input name = "in" type = "float" nodename = "_Boolean" />
    </dot>
    <dot name = "_14" type = "float">
        <input name = "in" type = "float" nodename = "_21" />
    </dot>
    <dot name = "_15" type = "float">
        <input name = "in" type = "float" nodename = "_22" />
    </dot>
    <dot name = "_16" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_17" />
    </dot>
    <combine3 name = "_17" type = "vector3">
        <input name = "in1" type = "float" nodename = "_12" />
        <input name = "in2" type = "float" nodename = "_13" />
        <input name = "in3" type = "float" nodename = "_14" />
    </combine3>
    <dot name = "_18" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_19" />
    </dot>
    <combine2 name = "_19" type = "vector2">
        <input name = "in1" type = "float" nodename = "_12" />
        <input name = "in2" type = "float" nodename = "_13" />
    </combine2>
    <constant name = "_Float" type = "float">
        <input name = "value" type = "float" value = "0.2" />
    </constant>
    <constant name = "_Boolean" type = "float">
        <input name = "value" type = "float" value = "1" />
    </constant>
    <constant name = "_Vector2" type = "vector2">
        <input name = "value" type = "vector2" value = "1,1" />
    </constant>
    <dot name = "_20" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_Vector2" />
    </dot>
    <extract name = "_21" type = "float">
        <input name = "in" type = "vector2" nodename = "_20" />
        <input name = "index" type = "integer" value = "0" />
    </extract>
    <extract name = "_22" type = "float">
        <input name = "in" type = "vector2" nodename = "_20" />
        <input name = "index" type = "integer" value = "1" />
    </extract>
    <constant name = "_Color" type = "color4">
        <input name = "value" type = "color4" value = "0,0.2530122,1,0" />
    </constant>
    <dot name = "_23" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_24" />
    </dot>
    <transformmatrix name = "_24" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_25" />
        <input name = "mat" type = "matrix44" nodename = "_26" />
    </transformmatrix>
    <dot name = "_25" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_35" />
    </dot>
    <transpose name = "_26" type = "matrix44">
        <input name = "in" type = "matrix44" nodename = "_27" />
    </transpose>
    <dot name = "_27" type = "matrix44">
        <input name = "in" type = "matrix44" nodename = "_Matrix4x4" />
    </dot>
    <mix name = "_28" type = "vector3">
        <input name = "bg" type = "vector3" nodename = "_6" />
        <input name = "fg" type = "vector3" nodename = "_36" />
        <input name = "mix" type = "float" nodename = "_37" />
    </mix>
    <multiply name = "_29" type = "vector3">
        <input name = "in1" type = "vector3" nodename = "_16" />
        <input name = "in2" type = "vector3" nodename = "_28" />
    </multiply>
    <GeometryModification name = "_30" type = "vertexshader">
    </GeometryModification>
    <realitykit_pbr name = "_31" type = "surfaceshader">
        <input name = "normal" type = "vector3" nodename = "_29" />
        <input name = "metallic" type = "float" nodename = "_38" />
        <input name = "emissiveColor" type = "color3" nodename = "_40" />
        <input name = "ambientOcclusion" type = "float" nodename = "_41" />
        <input name = "clearcoat" type = "float" nodename = "_42" />
        <input name = "baseColor" type = "color3" nodename = "_44" />
        <input name = "specular" type = "float" nodename = "_33" />
        <input name = "roughness" type = "float" nodename = "_32" />
        <input name = "clearcoatRoughness" type = "float" nodename = "_34" />
    </realitykit_pbr>
    <surfacematerial name = "USD_AllIO" type = "material">
        <input name = "vertexshader" type = "vertexshader" nodename = "_30" />
        <input name = "surfaceshader" type = "surfaceshader" nodename = "_31" />
    </surfacematerial>
    <subtract name = "_32" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" nodename = "_39" />
    </subtract>
    <dotproduct name = "_33" type = "float">
        <input name = "in1" type = "vector3" value = "0.299,0.587,0.114" />
        <input name = "in2" type = "vector3" nodename = "_29" />
    </dotproduct>
    <subtract name = "_34" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" nodename = "_43" />
    </subtract>
    <swizzle name = "_35" type = "vector4">
        <input name = "in" type = "color4" nodename = "_Color" />
        <input name = "channels" type = "string" value = "rgba" />
    </swizzle>
    <swizzle name = "_36" type = "vector3">
        <input name = "in" type = "vector4" nodename = "_1" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
    <swizzle name = "_37" type = "float">
        <input name = "in" type = "vector4" nodename = "_23" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_38" type = "float">
        <input name = "in" type = "vector3" nodename = "_29" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_39" type = "float">
        <input name = "in" type = "vector3" nodename = "_29" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_40" type = "color3">
        <input name = "in" type = "vector3" nodename = "_29" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
    <swizzle name = "_41" type = "float">
        <input name = "in" type = "vector3" nodename = "_29" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_42" type = "float">
        <input name = "in" type = "vector3" nodename = "_29" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_43" type = "float">
        <input name = "in" type = "vector3" nodename = "_29" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_44" type = "color3">
        <input name = "in" type = "vector3" nodename = "_29" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
</materialx>
