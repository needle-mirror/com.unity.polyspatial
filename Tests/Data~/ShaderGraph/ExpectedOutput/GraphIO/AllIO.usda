#usda 1.0
(
    metersPerUnit = 1
    upAxis = "Y"
)

def "MaterialX"
{
    def "Materials"
    {
        def Material "USD_AllIO"
        {
            matrix4d inputs:_Matrix4x4 = ((1, 0, 0, 0), (0, 1, 0, 0), (0, 0, 1, 0), (0, 0, 0, 1))
            float4 inputs:_Vector4 = (0, 1, 0, 0)
            float3 inputs:_Vector3 = (1, 0, 0)
            matrix3d inputs:_Matrix3x3 = ((1, 0, 0), (0, 1, 0), (0, 0, 1))
            float inputs:_Float = 0.2
            float inputs:_Boolean = 1
            float2 inputs:_Vector2 = (1, 1)
            color4f inputs:_Color = (0, 0.2530122, 1, 0)

            token outputs:mtlx:surface.connect = <realitykit_pbr.outputs:out>
            token outputs:realitykit:vertex.connect = <GeometryModifier.outputs:out>

            def Shader "realitykit_pbr"
            {
                uniform token info:id = "ND_realitykit_pbr_surfaceshader"
                float3 inputs:normal.connect = <../_18.outputs:out>
                float inputs:metallic.connect = <../_27.outputs:out>
                color3f inputs:emissiveColor.connect = <../_29.outputs:out>
                float inputs:ambientOcclusion.connect = <../_30.outputs:out>
                float inputs:clearcoat.connect = <../_31.outputs:out>
                color3f inputs:baseColor.connect = <../_33.outputs:out>
                float inputs:specular.connect = <../_22.outputs:out>
                float inputs:roughness.connect = <../_21.outputs:out>
                float inputs:clearcoatRoughness.connect = <../_23.outputs:out>
                token outputs:out
            }

            def Shader "GeometryModifier"
            {
                uniform token info:id = "ND_realitykit_geometrymodifier_vertexshader"
                token outputs:out
            }

            def Shader "_18"
            {
                uniform token info:id = "ND_multiply_vector3"
                float3 inputs:in1.connect = <../_9.outputs:out>
                float3 inputs:in2.connect = <../_17.outputs:out>
                float3 outputs:out
            }

            def Shader "_9"
            {
                uniform token info:id = "ND_dot_vector3"
                float3 inputs:in.connect = <../_10.outputs:out>
                float3 outputs:out
            }

            def Shader "_10"
            {
                uniform token info:id = "ND_combine3_vector3"
                float inputs:in1.connect = <../_5.outputs:out>
                float inputs:in2.connect = <../_6.outputs:out>
                float inputs:in3.connect = <../_7.outputs:out>
                float3 outputs:out
            }

            def Shader "_5"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../.inputs:_Float>
                float outputs:out
            }

            def Shader "_27"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_18.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_29"
            {
                uniform token info:id = "ND_swizzle_vector3_color3"
                float3 inputs:in.connect = <../_18.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_30"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_18.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_31"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_18.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_33"
            {
                uniform token info:id = "ND_swizzle_vector3_color3"
                float3 inputs:in.connect = <../_18.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_22"
            {
                uniform token info:id = "ND_dotproduct_vector3"
                float3 inputs:in1 = (0.299, 0.587, 0.114)
                float3 inputs:in2.connect = <../_18.outputs:out>
                float outputs:out
            }

            def Shader "_21"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1 = 1
                float inputs:in2.connect = <../_28.outputs:out>
                float outputs:out
            }

            def Shader "_28"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_18.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_23"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1 = 1
                float inputs:in2.connect = <../_32.outputs:out>
                float outputs:out
            }

            def Shader "_32"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_18.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_17"
            {
                uniform token info:id = "ND_mix_vector3"
                float3 inputs:bg.connect = <../_2.outputs:out>
                float3 inputs:fg.connect = <../_25.outputs:out>
                float inputs:mix.connect = <../_26.outputs:out>
                float3 outputs:out
            }

            def Shader "_26"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_16.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_16"
            {
                uniform token info:id = "ND_transformmatrix_vector4"
                float4 inputs:in.connect = <../_24.outputs:out>
                matrix4d inputs:mat.connect = <../.inputs:_Matrix4x4>
                float4 outputs:out
            }

            def Shader "_24"
            {
                uniform token info:id = "ND_swizzle_color4_vector4"
                color4f inputs:in.connect = <../.inputs:_Color>
                string inputs:channels = "rgba"
                float4 outputs:out
            }

            def Shader "_25"
            {
                uniform token info:id = "ND_swizzle_vector4_vector3"
                float4 inputs:in.connect = <../_1.outputs:out>
                string inputs:channels = "xyz"
                float3 outputs:out
            }

            def Shader "_1"
            {
                uniform token info:id = "ND_transformmatrix_vector4"
                float4 inputs:in.connect = <../.inputs:_Vector4>
                matrix4d inputs:mat.connect = <../.inputs:_Matrix4x4>
                float4 outputs:out
            }

            def Shader "_2"
            {
                uniform token info:id = "ND_transformmatrix_vector3"
                matrix3d inputs:mat.connect = <../.inputs:_Matrix3x3>
                float3 inputs:in.connect = <../.inputs:_Vector3>
                float3 outputs:out
            }

            def Shader "_6"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../.inputs:_Boolean>
                float outputs:out
            }

            def Shader "_7"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_14.outputs:out>
                float outputs:out
            }

            def Shader "_14"
            {
                uniform token info:id = "ND_extract_vector2"
                float2 inputs:in.connect = <../_13.outputs:out>
                int inputs:index = 0
                float outputs:out
            }

            def Shader "_13"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../.inputs:_Vector2>
                float2 outputs:out
            }
        }
    }
}
