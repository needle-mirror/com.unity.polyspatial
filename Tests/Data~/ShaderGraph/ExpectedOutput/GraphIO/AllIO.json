{"properties":[{"name":"_Matrix4x4","type":5},{"name":"_Vector4","type":3},{"name":"_Vector3","type":3},{"name":"_Matrix3x3","type":5},{"name":"_Float","type":1},{"name":"_Boolean","type":1},{"name":"_Vector2","type":3},{"name":"_Color","type":4}]}